From 2d99c7d829c6893da9c81b8a070b9dd9381858e5 Mon Sep 17 00:00:00 2001
From: Vikas Gupta <vikas.gupta@broadcom.com>
Date: Fri, 25 Apr 2014 13:05:48 +0530
Subject: [PATCH 0958/1532] xlp9xx: Change for macros on NAE-RX side.

	1) Number of calendar slots alloted to SGMII were little but high.
	2) Updated the parser seq-fifo size.
[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/netlogic/lib/netlib/include/netsoc_common.h | 11 ++++++++---
 arch/mips/netlogic/lib/netlib/src/netsoc_nae.c        |  2 +-
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/arch/mips/netlogic/lib/netlib/include/netsoc_common.h b/arch/mips/netlogic/lib/netlib/include/netsoc_common.h
index 1d239fd..5e1ebf8 100644
--- a/arch/mips/netlogic/lib/netlib/include/netsoc_common.h
+++ b/arch/mips/netlogic/lib/netlib/include/netsoc_common.h
@@ -72,9 +72,11 @@
 
 #define DEFAULT_XLP9XX_XAUI_IFACEFIFO_SZ	110
 #define DEFAULT_XLP9XX_XAUI_RXFIFO_SZ         	1888
-#define DEFAULT_XLP9XX_XAUI_PARSERSEQFIFO_SZ	480 
+#define DEFAULT_XLP9XX_XAUI_PARSERSEQFIFO_SZ	944  /*assuming 2048 ptp disabled.*/
 #define DEFAULT_XLP9XX_XAUI_NUM_FREEDESC	300
 
+#define DEFAULT_XLP9XX_SGMII_CAL_SLOTS         1
+
 #define DEFAULT_XLP2XX_XAUI_IFACEFIFO_SZ        110
 #define DEFAULT_XLP2XX_XAUI_RXFIFO_SZ           1888
 #define DEFAULT_XLP2XX_XAUI_PARSERSEQFIFO_SZ    480
@@ -258,8 +260,11 @@ static inline uint32_t get_numfree_desc(uint32_t intf_type)
 
 static inline int get_slots_required(int intf_type)
 {
-        if (intf_type == SGMII_IF)
-                return DEFAULT_SGMII_CAL_SLOTS;
+        if (intf_type == SGMII_IF){
+		if (is_nlm_xlp9xx())
+                		return DEFAULT_XLP9XX_SGMII_CAL_SLOTS;
+                	return DEFAULT_SGMII_CAL_SLOTS;
+		}
         else if (intf_type == XAUI_IF)
                 return DEFAULT_XAUI_CAL_SLOTS;
         else if (intf_type == RXAUI_IF)
diff --git a/arch/mips/netlogic/lib/netlib/src/netsoc_nae.c b/arch/mips/netlogic/lib/netlib/src/netsoc_nae.c
index 0094c74..ccb015f 100644
--- a/arch/mips/netlogic/lib/netlib/src/netsoc_nae.c
+++ b/arch/mips/netlogic/lib/netlib/src/netsoc_nae.c
@@ -1490,7 +1490,7 @@ void __netsoc_config_ingress_fifo(nae_t *nae)
 			}
 
 			size = ((lane == 0) ? nae->ports[port].prsr_seq_fifo_size : 0);
-			if ((IS_NLM_XLP9XX_FAMILY) && (nae->ptp1588_enabled == 0) && size)
+                       if ((IS_NLM_XLP9XX_FAMILY) && (nae->ptp1588_enabled == 0) && size && (!(nae->sgmii_complex_map)))
 				update_prsr_seq_fifo_size(nae, &size);
 
 			__netsoc_config_parserfifo(nae, nae->ports[port].hw_port_id + lane, cur_parser_base, size, 64, nae->ports[port].iftype);
-- 
1.9.1

